### ISSUE TITLE
Accessibility issue [tab key]

### ISSUE THREAD

**Issue by lalitha7** [cytoscape/cytoscape.js#3091]
We are using cytoscape bfs layout to display a graph, while doing accessibility testing using [Accessibility Insights for Web](https://accessibilityinsights.io/docs/web/overview/) extension, none of the nodes, links are reachable using tab key. This is a serious concern for accessibility as the interactive elements are not reachable through keyboard. Please let me know if there is any fix for this issue.

**Comment by nidhimittal90** [comment]
Hi team, is there any solution to reach nodes and connection of graph through keyboard ?

**Comment by maxkfranz** [comment]
Accessibility is an important issue, and it's great that you're considering it for your app.

An app must be designed itself with the accessibility features you’d like to support in mind, taking into account the app’s use cases and the user’s mental model.

Let's take tabbing as a simple example:

1. Do you allow tabbing through all the elements, or would users only want to tab through some of the elements?  Is it acceptable to force users to tab through potentially thousands of elements to get to the one they want?
2. In what order do you enable users to tab through elements?  By their geographic position?  By the graph topology?  Alphabetically?  By some app-specific semantics?
3. Do you force users to tab through elements based on where a Cytoscape instance just happens to exist in the DOM, or may the elements be tabbed through in a more suitable location?

These are but a few of the considerations you may make when designing your app w.r.t. accessibility.  So, as you can see, it's important to tailor your solutions based on what is going to give a good experience for the users of your app specifically.  Automated tools to assess accessibility, like Accessibility Insights for Web, are a good start, but they hardly paint the whole picture.

Generally, accessibility is something that needs to be built at the app level — especially because so many surrounding details depend on the design of the app itself.  Cytoscape generally shouldn't, and can't, make those decisions for you.

In future, we may publish some accessibility reading materials re. Cytoscape.  And since Cytoscape supports [extensions](https://js.cytoscape.org/#extensions), you're welcome to to publish reusable accessibility extensions for others who build similar apps to yours in future.

It's up to app authors like you to ensure that accessibility is properly addressed within the apps that they build.  It's great that you consider this issue to be important, and we encourage you to continue to consider it as you design and develop your app.

**Comment by lalitha7** [comment]
@maxkfranz Thanks for your detailed response. Yes, I do agree that there would be a number of ways the accessibility order be defined. 
Our use case has a graph with few nodes, and each holds some metadata which we are displaying on clicking the node. Since node can't be reached using tabs, the metadata is unreachable, hence the concern.

Let's say if we want to traverse the nodes in BFS layout in top-down order (left->right in same row). Could you please help us with the pointers on how to enable keyboard tab access to nodes in the first place? We shall try to work on defining the tab order as per our requirements.

Thanks for your help!
